Firebase/DynamicLinks/Public/FDLURLComponents.h

```diff
@@ -525,9 +528,42 @@ FIR_SWIFT_NAME(DynamicLinkComponents)
  * @param link Deep link to be stored in created Dynamic link. This link also called "payload" of
  *     the Dynamic link.
  * @param domain Domain of your App. This value must be equal to your assigned domain from Firebase
- *     Console.
+ *     Console. (e.g. xyz.page.link). Note that the domain scheme is required to be https and is
+ * assumed as such by this API.
  */
-- (instancetype)initWithLink:(NSURL *)link domain:(NSString *)domain;
+- (instancetype)initWithLink:(NSURL *)link
+                      domain:(NSString *)domain
+    DEPRECATED_MSG_ATTRIBUTE(
+        "This method is deprecated. Please use the new method with support for "
+        "domainURIPrefix- initWithLink:domainURIPrefix.");
+
+/**
+ * @method componentsWithLink:domain:
+ * @abstract Generates a Dynamic Link URL components object with the minimum necessary parameters
+ *     set to generate a fully-functional Dynamic Link.
+ * @param link Deep link to be stored in created Dynamic link. This link also called "payload" of
+ *     the Dynamic link.
+ * @param domainURIPrefix Domain URI Prefix of your App. This value must be either a. your assigned
+ * domain from the Firebase console or b. your custom domain or c. your custom domain with a valid
```

Comment:
Please change this documentation since custom domains aren't going to be available when this ships.

Reply:
Done. Removed all reference to custom domains.